# LSTM을 수행하기위해서 기존 데이터 형식 맞추기

In [17]:
# 환경변수 불러오기
import pandas as pd
from dotenv import load_dotenv
import os

load_dotenv()

True

In [18]:
# 경로 설정
file_path = os.getenv('FILE_PATH')
save_path = os.getenv('SAVE_PATH')

## 1. 종침후_TN_9시간_비교

In [19]:
try:
    tn = pd.read_csv(file_path+'lstm_data/sewageMeasurement/종침후_TN_9시간_비교.csv', encoding='utf-8')
except UnicodeDecodeError:
    try:
        tn = pd.read_csv(file_path+'lstm_data/sewageMeasurement/종침후_TN_9시간_비교.csv', encoding='cp949')
    except Exception as e:
        print(f"Error: {e}")

In [20]:
tn

_time_gateway  Sensor_S4_TN_9h  Unnamed: 2  Unnamed: 3  Unnamed: 4
0        9/1/22 0:00             8.00         NaN         NaN         NaN
1        9/1/22 0:10             7.97         NaN         NaN         NaN
2        9/1/22 0:20             8.25         NaN         NaN         NaN
3        9/1/22 0:30             8.01         NaN         NaN         NaN
4        9/1/22 0:40             7.95         NaN         NaN         NaN
...              ...              ...         ...         ...         ...
51915  8/31/23 23:10             8.05         NaN         NaN         NaN
51916  8/31/23 23:20             8.07         NaN         NaN         NaN
51917  8/31/23 23:30             8.06         NaN         NaN         NaN
51918  8/31/23 23:40             8.08         NaN         NaN         NaN
51919  8/31/23 23:50             8.01         NaN         NaN         NaN

[51920 rows x 5 columns]

In [21]:
# 필요한 열 제외 제거
tn = tn_1.drop(['Unnamed: 2', 'Unnamed: 3', 'Unnamed: 4'], axis=1)

In [22]:
tn

_time_gateway  Sensor_S4_TN_9h
0        9/1/22 0:00             8.00
1        9/1/22 0:10             7.97
2        9/1/22 0:20             8.25
3        9/1/22 0:30             8.01
4        9/1/22 0:40             7.95
...              ...              ...
51915  8/31/23 23:10             8.05
51916  8/31/23 23:20             8.07
51917  8/31/23 23:30             8.06
51918  8/31/23 23:40             8.08
51919  8/31/23 23:50             8.01

[51920 rows x 2 columns]

## 종침후_TN_9시간_비교.csv의 형식

_time_gateway Sensor_S4_TN_9h     
9/1/22 0:00, 8.00  
9/1/22 0:10, 7.97  
9/1/22 0:20, 8.25  

- datetime 인덱스를 사용하기위해 수정 필요

### _Time_gateway 변환

- YYYY/MM/DD HH:MM 형식

In [23]:
# 날짜 형식 변환 함수
def convert_date_format(date_str):
    return pd.to_datetime(date_str, format='%m/%d/%y %H:%M').strftime('%Y/%m/%d %H:%M')

In [24]:
# 날짜 형식 변환 적용
tn['_time_gateway'] = tn['_time_gateway'].apply(convert_date_format)

In [25]:
tn

_time_gateway  Sensor_S4_TN_9h
0      2022/09/01 00:00             8.00
1      2022/09/01 00:10             7.97
2      2022/09/01 00:20             8.25
3      2022/09/01 00:30             8.01
4      2022/09/01 00:40             7.95
...                 ...              ...
51915  2023/08/31 23:10             8.05
51916  2023/08/31 23:20             8.07
51917  2023/08/31 23:30             8.06
51918  2023/08/31 23:40             8.08
51919  2023/08/31 23:50             8.01

[51920 rows x 2 columns]

In [26]:
# 분석을 위한 데이터셋 저장
tn.to_csv(save_path+"LSTM_pipeline/Input/tn.csv",index=False)

## 2. 종침후_TOC_9시간_비교

In [28]:
try:
    toc = pd.read_csv(file_path+'lstm_data/sewageMeasurement/종침후_TOC_9시간_비교.csv', encoding='utf-8')
except UnicodeDecodeError:
    try:
        toc = pd.read_csv(file_path+'lstm_data/sewageMeasurement/종침후_TOC_9시간_비교.csv', encoding='cp949')
    except Exception as e:
        print(f"Error: {e}")

In [29]:
toc

_time_gateway  Sensor_S4_TOC_9h _time_gateway.1  \
0        9/1/22 0:00               7.0             NaN   
1        9/1/22 0:10               7.0             NaN   
2        9/1/22 0:20               7.1             NaN   
3        9/1/22 0:30               7.0             NaN   
4        9/1/22 0:40               6.9             NaN   
...              ...               ...             ...   
51915  8/31/23 23:10               5.9   8/31/23 23:10   
51916  8/31/23 23:20               5.8   8/31/23 23:20   
51917  8/31/23 23:30               5.9   8/31/23 23:30   
51918  8/31/23 23:40               5.9   8/31/23 23:40   
51919  8/31/23 23:50               5.8   8/31/23 23:50   

       pred_Sensor_S4_TOC_9h_xgboost  Unnamed: 4  Unnamed: 5  Unnamed: 6  \
0                                NaN         NaN         NaN         NaN   
1                                NaN         NaN         NaN         NaN   
2                                NaN         NaN         NaN         NaN   
3                                NaN         NaN         NaN         NaN   
4                                NaN         NaN         NaN         NaN   
...                              ...         ...         ...         ...   
51915                       5.482946         NaN         NaN         NaN   
51916                       5.476313         NaN         NaN         NaN   
51917                       5.465485         NaN         NaN         NaN   
51918                       5.455047         NaN         NaN         NaN   
51919                       5.438957         NaN         NaN         NaN   

       Unnamed: 7  Unnamed: 8  Unnamed: 9  Unnamed: 10  Unnamed: 11  \
0             NaN         NaN         NaN          NaN          NaN   
1             NaN         NaN         NaN          NaN          NaN   
2             NaN         NaN         NaN          NaN          NaN   
3             NaN         NaN         NaN          NaN          NaN   
4             NaN         NaN         NaN          NaN          NaN   
...           ...         ...         ...          ...          ...   
51915         NaN         NaN         NaN          NaN          NaN   
51916         NaN         NaN         NaN          NaN          NaN   
51917         NaN         NaN         NaN          NaN          NaN   
51918         NaN         NaN         NaN          NaN          NaN   
51919         NaN         NaN         NaN          NaN          NaN   

       Unnamed: 12  Unnamed: 13  Unnamed: 14  
0              NaN          NaN          NaN  
1              NaN          NaN          NaN  
2              NaN          NaN          NaN  
3              NaN          NaN          NaN  
4              NaN          NaN          NaN  
...            ...          ...          ...  
51915          NaN          NaN          NaN  
51916          NaN          NaN          NaN  
51917          NaN          NaN          NaN  
51918          NaN          NaN          NaN  
51919          NaN          NaN          NaN  

[51920 rows x 15 columns]

In [30]:
# 필요한 열 제외 제거
toc = toc[['_time_gateway', 'Sensor_S4_TOC_9h']]

In [31]:
toc

_time_gateway  Sensor_S4_TOC_9h
0        9/1/22 0:00               7.0
1        9/1/22 0:10               7.0
2        9/1/22 0:20               7.1
3        9/1/22 0:30               7.0
4        9/1/22 0:40               6.9
...              ...               ...
51915  8/31/23 23:10               5.9
51916  8/31/23 23:20               5.8
51917  8/31/23 23:30               5.9
51918  8/31/23 23:40               5.9
51919  8/31/23 23:50               5.8

[51920 rows x 2 columns]

## 종침후_TOC_9시간_비교.csv의 형식

_time_gateway Sensor_S4_TN_9h     
9/1/22 0:00, 7.0  
9/1/22 0:10, 7.0  
9/1/22 0:20, 7.1  

- datetime 인덱스를 사용하기위해 수정 필요

### _Time_gateway 변환

- YYYY/MM/DD HH:MM 형식

In [32]:
# 날짜 형식 변환 함수
def convert_date_format(date_str):
    return pd.to_datetime(date_str, format='%m/%d/%y %H:%M').strftime('%Y/%m/%d %H:%M')

In [33]:
# 날짜 형식 변환 적용
toc['_time_gateway'] = toc['_time_gateway'].apply(convert_date_format)

/var/folders/gp/txd8895x3vbc94rnxd7b3gjw0000gn/T/ipykernel_78280/4250179273.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  toc['_time_gateway'] = toc['_time_gateway'].apply(convert_date_format)


In [34]:
toc

_time_gateway  Sensor_S4_TOC_9h
0      2022/09/01 00:00               7.0
1      2022/09/01 00:10               7.0
2      2022/09/01 00:20               7.1
3      2022/09/01 00:30               7.0
4      2022/09/01 00:40               6.9
...                 ...               ...
51915  2023/08/31 23:10               5.9
51916  2023/08/31 23:20               5.8
51917  2023/08/31 23:30               5.9
51918  2023/08/31 23:40               5.9
51919  2023/08/31 23:50               5.8

[51920 rows x 2 columns]

In [35]:
# 분석을 위한 데이터셋 저장
toc.to_csv(save_path+"LSTM_pipeline/Input/toc.csv",index=False)